modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFileContigs.pm

```diff
@@ -0,0 +1,150 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+ Bio::EnsEMBL::Production::Pipeline::Webdatafile::WebdataFileContigs;
+
+=head1 DESCRIPTION
+  Compute Contigs for webdatafile dumps
+
+=cut
+
+package Bio::EnsEMBL::Production::Pipeline::Webdatafile::WebdataFileContigs;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Registry;
+use base qw/Bio::EnsEMBL::Production::Pipeline::Common::Base/;
+use Path::Tiny qw(path);
+use Carp qw/croak/;
+use JSON qw/decode_json/;
+use Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::GenomeLookup;
+use Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::IndexBed;
+
+sub param_defaults {
+  my ($self) = @_;
+  return {
+    %{$self->SUPER::param_defaults},
+  };
+}
+
+
+sub run {
+
+  my ($self) = @_;
+  my $species = $self->param('species');
+  my $current_step = $self->param('current_step') ;
+  my $output = $self->param('output_path');
+  my $app_path = $self->param('app_path'); 
+  my $genome_data = {
+    dbname     => $self->param('dbname'),
+    gca        => $self->param('gca'),
+    genome_id  => $self->param('genome_id'),
+    species    => $self->param('species'),
+    version    => $self->param('version'),
+    type       => $self->param('type'),  
+    root_path  => path($self->param('root_path'))
+  };
+  my $dba = Bio::EnsEMBL::Registry->get_DBAdaptor( $self->param('species'), $self->param('group') );
+  my $lookup = Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::GenomeLookup->new("genome_data" => $genome_data); 
+   my $slice_adaptor = Bio::EnsEMBL::Registry->get_adaptor( $self->param('species'), $self->param('group'), 'Slice' );
+  my $genome = $lookup->get_genome('1');
```

Comment:
Shoudn't it be $self->param('genome_id')?

Reply:
As analysis step run per species and we are creating Genome object for single species in the analysis i used constant value as 1 to get the genome object for current species . i will use species name which makes more readbility

Reply:
Ok fine.